### PR TITLE
Move sort filter card in the products screen to the RecyclerView

### LINKED ISSUE #3260: Products: “Filters” and “Sort” CTAs shouldn’t be sticky
**Describe the bug**

As @Garance91540 reported:

On scroll (on the products list), the “Filters” and “Sort” CTAs shouldn’t be sticky.

**To Reproduce**
1. Go to the Products tab.
2. Notice the "Filters" and "Sort" CTAs.
3. Scroll and notice those CTAs are sticky.

**Expected behavior**

**Mobile Environment**
Please include:
- Device: 
- Android version: 
- WooCommerce Android version: 5.6-rc-1

### PR DESCRIPTION
**Pull request by irfano**
Closes: #3260 

### Description

Sort-Filter card was sticky in the products screen. 
This PR moves Sort-Filter card in the RecyclerView, so it's not sticky anymore. If you touch Sort-Filter card, you couldn't scroll the list before. You can scroll by tapping Sort-Filter card now.

### Testing instructions

- Go to the Products tab.
- Scroll down the product list. You should have enough products to be able to scroll.
- Notice Sort-Filter cards are going up by scrolling.

### Images/gif

| before  | after |
| ------------- | ------------- |
| before products scrolled | after products scrolled |

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductItemDiffCallback.kt [hunk 1]
```diff
@@ -1,20 +0,0 @@
-package com.woocommerce.android.ui.products
-
-import androidx.recyclerview.widget.DiffUtil
-import com.woocommerce.android.model.Product
-
-object ProductItemDiffCallback : DiffUtil.ItemCallback<Product>() {
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductItemDiffCallback.kt:6]: `ProductItemDiffCallback` was common for `GroupedProductListAdapter` and `ProductListAdapter` before. But we need a different one for `ProductListAdapter`. Removed this class and added diff callbacks in each adapter class.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListAdapter.kt [hunk 2]
```diff
@@ -4,49 +4,157 @@ import android.view.LayoutInflater
...
+        }
+
+    private var sortFilterItem: ListItem.SortFilterItem? = null
+        set(value) {
+            val list = mutableListOf<ListItem>()
+
+            // Add sort filter item
+            if (value != null && value.show == true) {
+                list.add(value)
+            }
```
Comment (irfano) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListAdapter.kt:54]: We use `sortFilterItem`s `show` parameter to remove or add `sortFilterItem`. In some cases, we want to remove sortFilterItem view but want to keep the data to add the view again later.

File: WooCommerce/src/main/res/layout/fragment_product_list.xml [hunk 3]
```diff
@@ -19,23 +19,10 @@
...
-            android:id="@+id/products_sort_filter_card"
-            style="@style/Woo.Card"
-            android:layout_width="match_parent"
-            android:layout_height="wrap_content"
-            android:layout_marginBottom="@dimen/minor_00"
-            android:visibility="gone"
-            app:layout_constraintEnd_toEndOf="parent"
-            app:layout_constraintStart_toStartOf="parent"
-            app:layout_constraintTop_toBottomOf="@+id/products_wip_card"
+            app:cardElevation="@dimen/plane_00"
```
Comment (irfano) [inline WooCommerce/src/main/res/layout/fragment_product_list.xml:25]: Before, elevations of `FeatureWIPNoticeCard` and  `ProductSortAndFiltersCard` were the same. After adding `FeatureWIPNoticeCard` to RecyclerView, their elevations are not the same anymore, and there is a shadow between these cards. So, I set `FeatureWIPNoticeCard` elevation to `plane_00` to remove the shadow.

**Comment by irfano** [comment]
@hichamboushaba, do you think I should open another issue for making the new features card not sticky? In #3260 only sort-filter card is mentioned but in the iOS version, the new features card is not sticky either.
Screenshot_20220414_135429

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListAdapter.kt [hunk 4]
```diff
@@ -4,49 +4,156 @@ import android.view.LayoutInflater
...
+    fun updateSortFilterItem(
+        title: String? = sortFilterItem?.title,
+        show: Boolean? = sortFilterItem?.show,
+        filterCount: Int? = sortFilterItem?.filterCount
+    ) {
+        sortFilterItem = if (sortFilterItem == null) {
+            ListItem.SortFilterItem(title, show, filterCount)
+        } else {
+            sortFilterItem?.copy(title = title, show = show, filterCount = filterCount)
+        }
```
Comment (hichamboushaba) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductListAdapter.kt]: np, WDYT about replacing this with a shorter version: 
```kotlin
        sortFilterItem = sortFilterItem?.copy(title = title, show = show, filterCount = filterCount)
            ?: ListItem.SortFilterItem(title, show, filterCount)
```

**Review by hichamboushaba — APPROVED** [review]
Nice work @irfano 👍, I left just one np comment, but feel free to ignore it if you wish so.

**Comment by hichamboushaba** [comment]
> @hichamboushaba, do you think I should open another issue for making the new features card not sticky? In https://github.com/woocommerce/woocommerce-android/issues/3260 only sort-filter card is mentioned but in the iOS version, the new features card is not sticky either.

I think this should be a design team decision, the feedback banner is sticky in all of the other screens, so if there is need to change the behavior, then it has to be consistent for all screens.
If you have free time and want to follow up this subject, then you can bring with the design team, otherwise I don't think it's a major topic, and surely it can wait.

**Pull request merged**